workers/task-driver/src/driver.rs

```diff
@@ -201,49 +194,89 @@ impl TaskExecutor {
         Ok(())
     }
 
+    /// Handle a request to run a task immediately
+    pub fn handle_run_immediate(
+        &self,
+        wallet_id: WalletIdentifier,
+        task_id: TaskIdentifier,
+        task: TaskDescriptor,
+    ) -> Result<(), TaskDriverError> {
+        // Check if any non-preemptable tasks conflict with this task
+        if let Some(conflicting_task) = self.state().current_committed_task(&wallet_id)? {
+            log::error!(
+                "run immediate conflicts with committed task {conflicting_task:?}, aborting..."
+            );
+            return Ok(());
+        }
+
+        // Pause the queue for the wallet
+        todo!("implement task queue pausing");
+
+        // Start the task optimistically assuming that the queue is paused
+        self.spawn_task_on_runtime(task_id, task);
+
+        // Unpause the queue for the wallet
+        todo!("implement task queue unpausing");
+    }
+
+    /// Spawn the given task on the runtime
+    fn spawn_task_on_runtime(&self, task_id: TaskIdentifier, descriptor: TaskDescriptor) {
```

Comment:
nit: we can have `spawn_task_on_runtime` return a `Result<(), TaskDriverError>` to match the signatures of `handle_run_immediate` and `handle_notification_request` and make use of the existing logging block after the match statement in `run`

Reply:
it doesn't error